packages/@aws-cdk/aws-cloudformation-codepipeline/lib/pipeline-action.ts

```diff
@@ -0,0 +1,318 @@
+import codepipeline  = require('@aws-cdk/aws-codepipeline');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+
+// tslint:disable:max-line-length
+/**
+ * Properties common to all CloudFormation actions
+ */
+export interface CloudFormationCommonProps {
+    /**
+     * The name of the stack to apply this action to
+     */
+    stackName: string;
+
+    /**
+     * A name for the filename in the output artifact to store the AWS CloudFormation call's result.
+     *
+     * The file will contain the result of the call to AWS CloudFormation (for example
+     * the call to UpdateStack or CreateChangeSet).
+     *
+     * AWS CodePipeline adds the file to the output artifact after performing
+     * the specified action.
+     *
+     * @default No output artifact generated
+     */
+    outputFileName?: string;
+
+    /**
+     * The name of the output artifact to generate
+     *
+     * Only applied if `outputFileName` is set as well.
+     *
+     * @default Automatically generated artifact name.
+     */
+    outputArtifactName?: string;
+}
+
+/**
+ * Base class for Actions that execute CloudFormation
+ */
+export abstract class CloudFormationAction extends codepipeline.DeployAction {
+    /**
+     * Output artifact containing the CloudFormation call response
+     *
+     * Only present if configured by passing `outputFileName`.
+     */
+    public artifact?: codepipeline.Artifact;
+
+    constructor(parent: codepipeline.Stage, id: string, props: CloudFormationCommonProps, configuration?: any) {
+        super(parent, id, 'CloudFormation',  { minInputs: 0, maxInputs: 10, minOutputs: 0, maxOutputs: 1 }, {
+            ...configuration,
+            StackName: props.stackName,
+            OutputFileName: props.outputFileName,
+        });
+
+        if (props.outputFileName) {
+            this.artifact = this.addOutputArtifact(props.outputArtifactName || (this.parent!.name + this.name + 'Artifact'));
```

Comment:
Why use `this.parent!` when you have the guaranteed non-null `parent` available?

Reply:
I mean why can't you use `parent` instead of `this.parent!`?

Reply:
Oh. Well. I guess I could :)